src/scrapers/base-scraper-with-browser.ts

```diff
@@ -169,6 +131,57 @@ class BaseScraperWithBrowser<TCredentials extends ScraperCredentials> extends Ba
     });
   }
 
+  private async initializePage() {
+    debug('initialize browser page');
+    if ('browserContext' in this.options) {
+      debug('Using the browser context provided in options');
+      return this.options.browserContext.newPage();
+    }
+
+    if ('browser' in this.options) {
+      debug('Using the browser instance provided in options');
+      const { browser } = this.options;
+
+      /**
+       * For backward compatibility, we will close the browser even if we didn't create it
+       */
+      if (!this.options.skipCloseBrowser) {
+        this.cleanups.push(async () => {
+          debug('closing the browser');
+          await browser.close();
+        });
```

Comment:
If the browser exists and you run multiple accounts, this cleanup task will be pushed for each account (except the first one):
1. It will run the same cleanup multiple times
1. If only one account triggered, is it will close the browser?

Reply:
The logic here will run from `scrape > initialize` for each scrape, and we will have one cleanup call.

This is the exact same behavior as the current [`terminate` implementation](https://github.com/eshaham/israeli-bank-scrapers/blob/26837defff377f336b246117f715a20923e8c12a/src/scrapers/base-scraper-with-browser.ts#L277C2-L281C32) that is:

```
    if (!this.browser) {
      return;
    }

    await this.browser.close();
```